code/__DEFINES/traits.dm

```diff
@@ -84,6 +84,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_NEARSIGHT			"nearsighted"
 #define TRAIT_FAT				"fat"
 #define TRAIT_HUSK				"husk"
+#define TRAIT_GHOUL				"ghoul"
```

Comment:
Traits are meant to be generic properties.
Could you describe me what a ghoul trait is about?

Reply:
changes your appearance to be like that of a husk while preventing you from using eldritch magics.

Reply:
I see it shares a lot of checks with `TRAIT_HUSK`
What we want to do here is, instead of checking both, to separate the traits into more base components.

Instead of `if(HAS_TRAIT(src, TRAIT_HUSK) || HAS_TRAIT(src,TRAIT_GHOUL))` several times, make a trait both husks and ghouls share, and apply it on both, relating to the visual appearance. Then make other traits relating to the other generic characteristics.

Instead of `(is-complex-feature-A || is-complex-feature-B)` we want to do `(has-simple-property-that-both-A-and-B-share)` checks.

Reply:
ah i see, so instead of trait_husk and trait_ghoul i can make a trait trait_husk_visual ?

Reply:
eh i just delted the trait husk. the ghouling process will just make you one. not really that big of a deal